Automattic/wp-calypso
# Social Sign Up: Warn the user about an already existing account

#### Proposed Changes

We want to display a message when the user tries to socially authenticate using an e-mail that is already registered in WPCOM. That already happens in login, but it's not the case in the signup flow.

Well, it "kind of" happens: we're redirecting the user immediately, not giving them the opportunity to try again and select a different profile. And that process is buggy as (1) it displays the domains step for about a second and (2) it's not setting the e-mail, rendering a blank page so the user needs to re-type it.

That's subpar and we thought of a better experience: tell the user of the problem and give them a link so they could try it _if they want to_. That's better because it respects user preferences.

#### Testing Instructions

1. See "Testing Instructions" in https://github.com/Automattic/wp-calypso/pull/64957 to set up a working wpcalypso.wordpress.com environment on your local machine;
2. Create a WPCOM account with a Google e-mail with the classic e-mail and password method (NOT socially authenticating);

Open a new window, go to https://wpcalypso.wordpress.com/start/user and click **Continue with Google** choosing the same e-mail you've used to create the account. After choosing the account in the pop-up, you should see this screen: 

<img width="1241" alt="image" src="https://user-images.githubusercontent.com/36432/180775890-d200e552-a0e0-491f-b611-430bebe12086.png">

Clicking on the link should redirect you to the login page, with the e-mail being prefilled and form representing the social linking state.

<img width="1240" alt="image" src="https://user-images.githubusercontent.com/36432/180791852-cf7f0836-8f7b-4c8a-940a-c7c063a838cc.png">

Now, open https://wordpress.com/me/security/social-login and socially connect with Google there. After that's done, when you try to sign in with Google at https://wpcalypso.wordpress.com/start/user, you should be redirected to `/start/domains` instead of being alerted about an existing user.

Related to https://github.com/Automattic/wp-calypso/issues/65654.
